### ISSUE TITLE
NaN losses (depth/offset3d/size3d/heading) suddenly appear at epoch 7, training crashes (possible gradient explosion)

### ISSUE THREAD

**Issue by zjack2024-g** [HyeonjeongPark37/MonoMH#1]
Thank you for the excellent work in this WACV paper!

When I try to reproduce it, I encounter a gradient explosion problem.
## Problem Description
Training is stable for the first 6 epochs with losses gradually decreasing, but starting from **epoch 7**, several regression losses suddenly become **NaN**, causing the entire training to crash.

Key log snippet at the beginning of epoch 7:
2026-03-03 10:48:29,760 INFO ------ TRAIN EPOCH 007 ------
2026-03-03 10:48:29,761 INFO Learning Rate: 0.001250
2026-03-03 10:48:30,129 INFO Weights: depth_:0.0075, heading_:0.0103, offset2d_:1.0000, offset3d_:0.0103, seg_:1.0000, size2d_:1.0000, size3d_:0.0103,
badNAN----------------depth_loss tensor(nan, device='cuda:0', grad_fn=<AddBackward0>)
badNAN----------------offset3d_loss tensor(nan, device='cuda:0', grad_fn=<MeanBackward0>)
badNAN----------------size3d_loss tensor(nan, device='cuda:0', grad_fn=<MeanBackward0>)
badNAN----------------heading_loss tensor(nan, device='cuda:0', grad_fn=<AddBackward0>)

For comparison, an example from epoch 1 (normal behavior):
2026-03-03 10:54:14,089 INFO BATCH[0100/0116] depth_loss:33.4890, heading_loss:3.7329, offset2d_loss:1.2356, offset3d_loss:1.0381, pixel_wise_depth_loss:30.3752, roi_wise_depth_loss:3.1137, seg_loss:11.8521, size2d_loss:21.1401, size3d_loss:0.5619,

## Questions
Have you / other users ever encountered NaN suddenly appearing around epoch 7 on these regression losses?

Happy to provide more logs, config yaml, or even a minimal repro.

Thank you for your time and great work on this repo!

**Comment by jiashengsu418** [comment]
我也遇到了这个问题，是不是数据增强的策略有问题，我把数据增强都关掉就好了，但是这应该不是正确的解决办法
